src/dmd/lexer.d

```diff
@@ -1955,6 +1969,11 @@ class Lexer : ErrorHandler
             error("integer overflow");
             err = true;
         }
+        if ((base == 2 && !anyBinaryDigitsUS) ||
+            (base == 16 && !anyHexDigitsNoSingleUS))
+        {
+            deprecation("Invalid integer literal is used, use valid one");
```

Comment:
Maybe mention the name of the literal here?
E.g.

> `0x` isn't a valid integer literal

Reply:
Please include what literal to use to get the current behavior.